flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java

```diff
@@ -75,32 +92,584 @@ public void clean() {
+    explainNoPushDown = getTableEnv().explainSql(sqlNoPushDown);
+    Assert.assertFalse("Explain should not contain FilterPushDown",
+        explainNoPushDown.contains(expectedFilterPushDownExplain));
+    sqlNoPushDown = "SELECT * FROM  " + TABLE_NAME + "  WHERE data LIKE '%%ice%%g' ";
+    resultLike = sql(sqlNoPushDown);
+    Assert.assertEquals("Should have 1 record", 1, resultLike.size());
+    Assert
+        .assertArrayEquals("Should produce the expected record", new Object[] {1, "iceberg", 10.0}, resultLike.get(0));
+    Assert.assertEquals("Should not push down a filter", Expressions.alwaysTrue(), lastScanEvent.filter());
+
+    sqlNoPushDown = "SELECT * FROM  " + TABLE_NAME + "  WHERE data LIKE '%' ";
+    explainNoPushDown = getTableEnv().explainSql(sqlNoPushDown);
+    Assert.assertFalse("Explain should not contain FilterPushDown",
+        explainNoPushDown.contains(expectedFilterPushDownExplain));
+    sqlNoPushDown = "SELECT * FROM  " + TABLE_NAME + "  WHERE data LIKE '%%' ";
+    resultLike = sql(sqlNoPushDown);
+    Assert.assertEquals("Should have 2 records", 2, resultLike.size());
+    List<Object[]> expectedRecords = Lists.newArrayList();
+    expectedRecords.add(new Object[] {1, "iceberg", 10.0});
+    expectedRecords.add(new Object[] {2, "b", 20.0});
+    Assert.assertArrayEquals("Should produce the expected record", expectedRecords.toArray(), resultLike.toArray());
+    Assert.assertEquals("Should not push down a filter", Expressions.alwaysTrue(), lastScanEvent.filter());
+
+    sqlNoPushDown = "SELECT * FROM  " + TABLE_NAME + "  WHERE data LIKE 'iceber_' ";
+    explainNoPushDown = getTableEnv().explainSql(sqlNoPushDown);
+    Assert.assertFalse("Explain should not contain FilterPushDown",
+        explainNoPushDown.contains(expectedFilterPushDownExplain));
+    resultLike = sql(sqlNoPushDown);
+    Assert.assertEquals("Should have 1 record", 1, resultLike.size());
+    Assert
+        .assertArrayEquals("Should produce the expected record", new Object[] {1, "iceberg", 10.0}, resultLike.get(0));
+    Assert.assertEquals("Should not push down a filter", Expressions.alwaysTrue(), lastScanEvent.filter());
+  }
+
+  @Test
+  public void testFilterPushDown2Literal() {
+    String sql2Literal = String.format("SELECT * FROM %s WHERE 1 > 0 ", TABLE_NAME);
+    String explain2Literal = getTableEnv().explainSql(sql2Literal);
+    Assert.assertFalse("Explain should not contain FilterPushDown",
+        explain2Literal.contains(expectedFilterPushDownExplain));
+    Assert.assertNull("Should not push down a filter", lastScanEvent);
+  }
+
+  /**
+   * NaN is not supported by flink now, so we add the test case to assert the parse error, when we upgrade the flink
+   * that supports NaN, we will delele the method, and add some test case to test NaN.
+   */
+  @Test
+  public void testSqlParseError() {
+    String sqlParseErrorEqual = String.format("SELECT * FROM %s WHERE d = DOUBLE('NaN') ", TABLE_NAME);
```

Comment:
I don't see the function `DOUBLE` documented in Flink's built-in functions. I think what you want is to use `CAST` instead: `CAST('NaN' AS DOUBLE)`.